src/librustc_mir/interpret/const_eval.rs

```diff
@@ -376,6 +376,25 @@ impl<'mir, 'tcx> super::Machine<'mir, 'tcx> for CompileTimeEvaluator {
                 };
                 ecx.write_scalar(out_val, dest)?;
             }
+            "overflowing_add" | "overflowing_sub" | "overflowing_mul" => {
+                // ToDo add *_with_overflow and make them const in libcore/num/mods.rs
+                let ty = substs.type_at(0);
+                let layout = ecx.layout_of(ty)?;
+                let lhs = ValTy {
+                    value: Value::Scalar(ecx.read_scalar(args[0])?),
+                    layout
+                };
+                let rhs = ValTy {
+                    value: Value::Scalar(ecx.read_scalar(args[1])?),
+                    layout
+                };
+                match intrinsic_name {
+                    "overflowing_add" => ecx.binop_ignore_overflow(BinOp::Add, lhs, rhs, dest)?,
```

Comment:
You can simplify this by having each brach only be the `BinOp::Foo` element and storing the entire match in a variable. Then, below the match, only have a single `binop_ignore_overflow` call